Only show audio message when track ID matches active

Anuj012004 opened the pull request:
<!-- prettier-ignore -->
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, please consider opening one before creating this pull request. -->

Fixes #2202  by @obulat 

## Description
An audio-related error is observed across all audio cells in the content view. Specifically, all cells incorrectly display an error message regardless of their state. This occurs because the application does not validate whether the audio.id matches the activeMediaStore.id when setting the message. As a result, messages intended for the active audio track are incorrectly shown for all audio tracks.


## Testing Instructions
<!-- Give steps for the reviewer to verify that this PR fixes the problem; or delete this section entirely. -->
A reliable way to test this behavior is to find `.oga` audio files in Safari.
1. Run the app locally and go to http://localhost:8443/search/audio?q=cat&extension=oga
2. Click on Play: you should see the error message only on the active track.
<img width="688" alt="Screenshot 2025-01-16 at 1 26 59 PM" src="https://github.com/user-attachments/assets/d54094c1-a32a-43a1-afb4-4b9db2f0ffa8" />

Compare with the same in production: you see that the error message is displayed on every track.
<img width="690" alt="Screenshot 2025-01-16 at 1 27 33 PM" src="https://github.com/user-attachments/assets/ed3313cc-d673-4d37-a7ef-b9ea9749f9ed" />


## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. -->

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

dhruvkb commented on frontend/src/components/VAudioTrack/VAudioTrack.vue:
```diff
@@ -296,9 +296,9 @@ watch(
-    ? t(`audioTrack.messages.${activeMediaStore.message}`)
+const message = computed(() => 
+  activeMediaStore.id === props.audio.id && activeMediaStore.message 
+    ? i18n.t(`audioTrack.messages.${activeMediaStore.message}`).toString() 
```
@obulat's comment https://github.com/WordPress/openverse/issues/2202#issuecomment-1635406889, although correct in terms of the changes, was a bit out of date. We can access `t` directly as it has been defined in line 70. Also there is no longer a need to use `.toString()`.

```suggestion
    ? t(`audioTrack.messages.${activeMediaStore.message}`)
```

In fact, using `i18n` introduces an error because that's an undefined variable.

```
Error: frontend types: src/components/VAudioTrack/VAudioTrack.vue(301,7): error TS2304: Cannot find name 'i18n'.
```

dhruvkb commented on frontend/src/components/VAudioTrack/VAudioTrack.vue:
```diff
@@ -296,9 +296,9 @@ watch(
-const message = computed(() =>
-  activeMediaStore.message
-    ? t(`audioTrack.messages.${activeMediaStore.message}`)
+const message = computed(() => 
+  activeMediaStore.id === props.audio.id && activeMediaStore.message 
```
Remove trailing spaces. 

> [!TIP]
> This can be done automatically by setting up the pre-commit and pre-push Git hooks > or by running `ov just lint` manually.

```suggestion
const message = computed(() =>
  activeMediaStore.id === props.audio.id && activeMediaStore.message
```

Anuj012004 commented:
Hey @dhruvkb,

I've updated the PR based on your suggestions:

1.Removed .toString() as it's no longer needed.
2.Using t directly since it's defined on line 70.
3.Fixed whitespace issues.

Please review and let me know if any other changes are needed. Thanks for the helpful feedback!

Anuj012004 commented:
Hey @dhruvkb,

One check failed on my PR. Could you please take a look to help identify what needs to be fixed?
 Thanks!

dhruvkb commented:
@Anuj012004 thanks for incorporating the feedback from the PR review. The changes in your PR are correct, and you might have noticed that the previously failing Lint check has passed. However the PR is failing the unit tests defined in the file `frontend/test/unit/specs/components/AudioTrack/v-audio-track.spec.js`. This means that the tests in that file have to be updated.

Now you would need to make changes to that file to rectify any incorrect test conditions, and run the tests locally with `just frontend/run test:unit`. Feel free to ask us if you need any help or assistance.

Anuj012004 commented:
Hey @dhruvkb, 
Please help me to fix the issue. I added test cases and due to my system's limitations I am unable to debug the failing tests effectively.

obulat commented on frontend/test/unit/specs/components/AudioTrack/v-audio-track.spec.js:
```diff
@@ -158,4 +158,33 @@ describe("AudioTrack", () => {
+    const { queryByText } = await render(VAudioTrack, options)
+    expect(queryByText(/playing/i)).not.toBeInTheDocument()
+  })
 })
```
```suggestion
})

```

obulat reviewed (approved):
Thank you for your contribution, @Anuj012004 ! 

I fixed the unit tests, and am going to merge this PR.
